lib/image-stream/parse-partitions.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const Bluebird = require('bluebird');
+const MBR = require('mbr');
+const GPT = require('gpt');
+
+const MAX_BYTES = 65536;
+const BLOCK_SIZE = 512;
+const INITIAL_LENGTH = 0;
+
+/**
+ * Attempt to read the MBR and GPT from an imagestream
+ * NOTE: This operation will consume the first `MAX_BYTES`
+ * of the stream and then destroy the stream.
```

Comment:
Nope, because the stream would be partially consumed before the method returns (or resolves), so that stream would be unusable even if left alive. It's safer to just open a new stream for a full read in most cases anyways.